docs/upgrading-to-aws-auth-kubernetes-provider.md

```diff
@@ -0,0 +1,14 @@
+# Upgrading from version <= 7.x to 8.0.0
```

Comment:
I think we move these notes to the changelog as this will be the first place people look for upgrade notes.

Historically we haven't written much more than `breaking` for things like this but I think we should start writing more detail in the changelog and this change is the perfect example 😃